src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java

```diff
@@ -0,0 +1,274 @@
+     * are set and the action button is configured based on the current book status.
+     *
+     * @param savedInstanceState an instance state that has the state of the BorrowerBookDetailsActivity
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_borrower_book_details);
+        getSupportActionBar().setTitle("Book Details");
+
+        Intent intent = getIntent(); // gets the previously created intent
+        book = (Book) intent.getSerializableExtra(ListingBooksActivity.EXTRA_BOOK);
+        user = (User) intent.getSerializableExtra(ListingBooksActivity.USER);
+
+        titleTextView = findViewById(R.id.borrower_book_details_title_text);
+        authorTextView = findViewById(R.id.borrower_book_details_author_text);
+        isbnTextView = findViewById(R.id.borrower_book_details_isbn_text);
+        descriptionTextView = findViewById(R.id.borrower_book_details_description_text);
+        imageView = findViewById(R.id.borrower_book_details_book_image);
+        ownedByTextView = findViewById(R.id.borrower_book_details_owned_by);
+        statusTextView = findViewById(R.id.borrower_book_details_book_status_text);
+
+        actionButton = findViewById(R.id.borrower_book_details_action_btn);
+
+        setBookDetails();
+        fillBookDetails();
+        configureActionButton();
+    }
+
+    /**
+     * This function handles retrieving data from the Book object
+     */
+    private void setBookDetails() {
+        isbn = book.getIsbn();
+        author = book.getAuthor();
+        title = book.getTitle();
+        description = book.getDescription();
+        ownedBy = book.getOwnerId().toString();
+        status = book.getStatus().toString();
+    }
+
+    /**
+     * This function handles filling the text fields and image with the book data
+     */
+    private void fillBookDetails() {
+        titleTextView.setText(title);
+        authorTextView.setText(author);
+        isbnTextView.setText("ISBN: " + isbn);
+        descriptionTextView.setText("Description: " + description);
+        // TODO: imageView.setImageBitmap();
+        ownedByTextView.setText("Owned by: " + ownedBy);
+        statusTextView.setText("Status: " + status);
+    }
+
+    /**
+     * This function handles setting the text of the action button and setting different
+     * onClickListeners of the button depending on the status of the book
+     */
+    private void configureActionButton() {
+        switch (book.getStatus()) {
+            case AVAILABLE:
+                actionButton.setText("Request");
+                actionButton.setOnClickListener(requestBookListener);
+                break;
+            case REQUESTED:
+                // TODO: set button style to OutlinedButton
+                actionButton.setText("Requested");
+                actionButton.setOnClickListener(aVoid -> {});
```

Comment:
anyone know of a way to programmatically set the button style to `Button.OutlinedButton`?

Reply:
ContextThemeWrapper newContext = new ContextThemeWrapper(baseContext, R.style.MyStyle);
button = new Button(newContext);

Reply:
also https://github.com/airbnb/paris

Reply:
https://ask.xiaolee.net/questions/1042508